server/sms/sms_connector.go

```diff
@@ -32,142 +32,140 @@ type Config struct {
 	Schema string
 }
 
-type conn struct {
+type gateway struct {
 	config *Config
+
 	sender Sender
 	router router.Router
-	logger *log.Entry
 	route  *router.Route
 
-	ctx        context.Context
-	cancel     context.CancelFunc
 	LastIDSent uint64
+
+	ctx        context.Context
+	cancelFunc context.CancelFunc
+
+	logger *log.Entry
 }
 
-func New(router router.Router, sender Sender, config Config) (*conn, error) {
+func New(router router.Router, sender Sender, config Config) (*gateway, error) {
 	if *config.Workers <= 0 {
 		*config.Workers = connector.DefaultWorkers
 	}
 	config.Schema = SMSSchema
 	config.Name = SMSDefaultTopic
-
-	gw := &conn{
-		router: router,
-		logger: logger.WithField("name", config.Name),
+	return &gateway{
 		config: &config,
+		router: router,
 		sender: sender,
-	}
-
-	return gw, nil
+		logger: logger.WithField("name", config.Name),
+	}, nil
 }
 
-func (c *conn) Start() error {
-	c.logger.Debug("Starting gateway")
+func (g *gateway) Start() error {
```

Comment:
The filename is `sms_connector.go` and inside we have a `gateway` struct.
Theoretically this is the connector. and in order to follow the other connectors the name should be `sms` for the struct.
And the gateway is nexmo.